lib/view/instruments_screen.dart

```diff
@@ -32,25 +35,46 @@ class _InstrumentsScreenState extends State<InstrumentsScreen> {
     }
   }
 
+  void _filterInstruments(String query) {
+    setState(() {
+      if (query.isEmpty) {
+        _filteredIndices =
+            List<int>.generate(instrumentHeadings.length, (index) => index);
+      } else {
+        _filteredIndices = [];
+        for (int i = 0; i < instrumentHeadings.length; i++) {
+          if (instrumentHeadings[i]
+              .toLowerCase()
+              .contains(query.toLowerCase())) {
+            _filteredIndices.add(i);
+          }
+        }
```

Comment:
```suggestion
        _filteredIndices = List.generate(instrumentHeadings.length, (i) => i)
    .where((i) => instrumentHeadings[i].toLowerCase().contains(query.toLowerCase()))
    .toList();
```

Let's use the full potential of Dart